doc/source/io.rst

```diff
@@ -4093,6 +4093,27 @@ The key functions are:
 
 .. _io.bigquery_reader:
 
+Authentication
+''''''''''''''
+
+You will be authenticated to the specified BigQuery account via Google's Oauth2 mechanism.
+
+Primary auth method is as simple as following the
+prompts in a browser window which will be opened for you. Should the browser not
+be available, or fail to launch, a code will be provided to complete the process
+manually.  Additional information on the authentication mechanism can be found
```

Comment:
I agree. It needs. But I'm not going to do it in this pull request. I'm just interested in support of service account auth and this is what I understand. I have just little knowledge of this primary auth method (user account).

If you're not satisfied with this 'Should the browser not be available, or fail to launch, a code will be provided to complete the process manually.', then I can remove it and reader will be redirected to google site.